lib/galaxy/tools/wrappers.py

```diff
@@ -3,8 +3,33 @@
 import shlex
 import tempfile
 from functools import total_ordering
+from typing import (
+    Any,
+    cast,
+    Dict,
+    Iterable,
+    Iterator,
+    KeysView,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    TYPE_CHECKING,
+    Union,
+)
 
 from galaxy import exceptions
+from galaxy.datatypes.registry import Registry
+from galaxy.jobs import ComputeEnvironment
+from galaxy.model import (
+    DatasetCollection,
+    DatasetCollectionElement,
+    DatasetCollectionInstance,
+    DatasetInstance,
+    HasTags,
+    HistoryDatasetCollectionAssociation,
+)
+from galaxy.model.metadata import MetadataCollection
```

Comment:
Likewise, these could also be moved under `if TYPE_CHECKING:`  to reduce the chances of cyclic imports in the future